src/material.cpp

```diff
@@ -1221,8 +1221,8 @@ pyne::comp_map pyne::Material::decay_heat() {
   double masspermole = mass * pyne::N_A;
   for (pyne::comp_iter i = comp.begin(); i != comp.end(); ++i) {
     dh[i->first] = masspermole * (i->second) * \
-                   decay_const(i->first) * q_val(i->first) / \
-                   atomic_mass(i->first) / pyne::MeV_per_MJ;
+                   decay_const(metastable_id(i->first,nucname::snum(i->first))) * \
+                   q_val(i->first) / atomic_mass(i->first) / pyne::MeV_per_MJ;
```

Comment:
Shouldn’t `q_val` also depend on the isomeric state?

Reply:
Yeah, `q_val` depends on the isomeric state but the table that holds the different q_values uses zas_id unlike the table for the decay constants which uses state_id